### ISSUE TITLE
2024.3 performance issues (seemingly) caused by Catppuccin?

### ISSUE THREAD

**Issue by Crocmagnon** [catppuccin/jetbrains#190]
### Is there an existing issue outlining your problem?

- [X] I have searched the existing issues and they do not solve my problem.

### Have you followed the instructions in the the pinned issue?

- [X] I have followed the instructions in the pinned issue and they do not solve my problem.

### Describe your problem.

When using alt+enter to trigger actions, the IDE (Goland) hangs forever. It shows as "unresponsive" in macOS process list and needs to be killed.

### Attach screenshots.

With Catppuccin theme plugin enabled and active.
After alt+enter, I'm trying to use arrows up+down but it doesn't move. My cursor also showed the "spinning ball", though it doesn't render on the screencast.
https://github.com/user-attachments/assets/1aac8640-7a3f-475c-8752-d6c92f8204fd

With Catppuccin theme plugin disabled
https://github.com/user-attachments/assets/892e1cfc-d363-435b-b64e-4b44ebea2afb

Looks like it also affects at least one other user (https://plugins.jetbrains.com/plugin/18682-catppuccin-theme/reviews):
<img width="896" alt="image" src="https://github.com/user-attachments/assets/c8dc2a15-86e4-4bc1-9b0f-2b2c51bba0a4">

### What IDE(s) are you seeing the problem on?

GoLand 2024.3 Build #GO-243.21565.208, built on November 14, 2024

### Any additional comments?

This doesn't happen with builtin themes as shown in the screencast.

Also, I really love Catppuccin themes! Thank you for making them available for so many platforms 🙏🏻

**Comment by sgoudham** [comment]
Hey 👋,
Thanks for taking the time to report. 

Personally, I must say that I am completely baffled at this. I'm still not convinced that Catppuccin is the problem because honestly I don't even know what we could change. Realistically, this plugin is only XML files that list hex codes which JetBrains is responsible for loading.

I suppose I'd be interested to know if the person on the plugin review was also using Goland or if it was another IDE.

Can you try invoking the context menu with another theme plugin enabled on the marketplace and see if you can reproduce this behaviour? Failing that, could you try to reproduce this in another IDE?

I'm struggling to think of the root cause because the context menus are fine in Intellij for me when using Catppuccin.

**Comment by Crocmagnon** [comment]
Looks like the issue happens more specifically when Catppuccin Latte is selected as the "Preferred Theme" (not the same as "Preferred Editor Color Scheme").

<img width="1181" alt="Capture d’écran 2024-11-30 à 20 15 34" src="https://github.com/user-attachments/assets/b878fab8-2b5c-4cce-bfaa-1c6d385f1534">

The effect seems stronger when both the theme and color scheme are set to Catppuccin but setting the theme to "Light" for example makes this a non-issue.

I tried several themes from the "Material Theme UI" but I couldn't reproduce.

I'm able to reproduce in PyCharm.
I was also able to reproduce the issue in a brand new RustRover without any other plugins - only Catppuccin theme. The screencast looks interesting:
https://github.com/user-attachments/assets/bf027879-9858-48c2-a4a0-777e294ecced

We can see that it wants to redraw the popup quite often, I wonder whether there is a rendering issue somewhere.

**Comment by sgoudham** [comment]
Can you downgrade your IDE version to 2024.2.x instead of using 2024.3 and see if this issue still persists?

**Comment by Crocmagnon** [comment]
The issue doesn't appear on GoLand 2024.2.4 or PyCharm 2024.2.5. After upgrading again to 2024.3, the issue re-appears instantly.

**Comment by sgoudham** [comment]
Okay... that's interesting. It looks like the issue can be isolated to versions after 2024.3 but only to the Catppuccin theme plugin? It's weird to have this type of regression but for (seemingly) only Catppuccin.

This plugin is only defining JSON and XML files so I cannot think of anything to change on our side at the moment.

For what its worth, a fair amount of users have reported performance issues on the [2024.3 release](https://blog.jetbrains.com/idea/2024/11/intellij-idea-2024-3/):

![image](https://github.com/user-attachments/assets/30c35dc9-8979-4e3b-be6d-21ef5cad7018)
![image](https://github.com/user-attachments/assets/932a99ca-c439-434b-a98f-a11ccd574bf8)

I don't have any explanations on why the issue is appearing for when only the Catppuccin theme is enabled though.

**Comment by Crocmagnon** [comment]
Oh well maybe the issue will sort itself out in a fix later? In the meantime I reverted to 2024.2 which seems to work fine.

Thank you for being so responsive. At least the issue is documented.
I’ll try to remember to check in here once fixés are published to 2024.3.

**Comment by sgoudham** [comment]
Yeah this one is very odd to me, thanks for taking the time to take the screen recordings.

I can't believe that a simple colourscheme plugin can be responsible for IDE freezes/hangs but I'm happy to merge a PR that fixes this if Catppuccin is indeed the problem. Unfortunately, I don't have the time to look into this since there's very little, if at all, to debug and investigate.

I'll keep this issue open since there was another person on the reviews that mentioned the IDE being laggy.

**Comment by Crocmagnon** [comment]
Searching for "catppuccin" in YouTrack surfaces multiple issues similar to the one I reported, and it looks like Catppuccin is not the only affected theme. Gerry Light was also mentioned several times.

* search: https://youtrack.jetbrains.com/issues?q=catppuccin
* IDEA freeze: https://youtrack.jetbrains.com/issue/IDEA-363387/IDE-freezes-randomly-for-several-seconds-since-update-to-2024.3
* RustRover freeze: https://youtrack.jetbrains.com/issue/RUST-16455/RustRover-freezes-and-not-responding-on-implement-missing-members-action-menu
* Webstorm freeze: https://youtrack.jetbrains.com/issue/WEB-70458/Webstorm-2024.3-freezes-when-showing-More-actions-when-a-particular-theme-is-selected
* IDE freeze: https://youtrack.jetbrains.com/issue/JBR-7693/IDE-freeze-on-macOS-at-sun.lwawt.macosx.CPlatformWindow.nativeDelayShowing

The upstream issue hypothesis seems to be confirmed.

I was able to reproduce the issue in IntelliJ IDEA 2024.3. They released a preview version of 2024.3.1 but unfortunately it doesn't fix the issue.

**Comment by SupianIDz** [comment]
@sgoudham This only happens with Catppuccin Latte.

**Comment by sgoudham** [comment]
> @sgoudham This only happens with Catppuccin Latte.

Skimming the YouTrack issues linked above, it seems like it's an issue with any theme that offers a Light theme, such as Gerry Light theme and Solarized Light.

I don't believe this plugin can do anything to fix it.

**Comment by Crocmagnon** [comment]
Linking to https://youtrack.jetbrains.com/issue/JBR-7981/Intellij-2024.3-freezes-continuously-on-opening-Context-Menu-Opt-Enter as well. Thanks @sgoudham for chiming in there.

**Comment by maxxscho** [comment]
I can confirm this. I'm having the same issues with Catppuccin and also with Gerry Light on PHPStorm since 2024.3.
Maybe this is some kind of render problem of this context menu on some themes. Something with box shadows or similar? Just guessing here…

**Comment by steebchen** [comment]
Yeah somehow my IJ always freezes ever half hour or so when doing basic things on the light theme.

**Comment by Erxhan** [comment]
I can confirm that it's only the `Catppuccin Latte` having this performance issue

**Comment by Crocmagnon** [comment]
Awesome, thanks!
Looks like jetbrains has a fix coming in 2025.1 EAP as well.
